src/app.js

```diff
@@ -1,31 +1,35 @@
-'use strict';
-
-/* istanbul ignore next */ if (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test')
-  require('dotenv').config(); // eslint-disable-line import/no-extraneous-dependencies
-
-const express = require('express');
+// eslint-disable-next-line import/no-extraneous-dependencies
+import dotenv from 'dotenv';
+import express from 'express';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+import Settings from './lib/settings.js';
+import * as addons from './lib/addons.js';
+import { logger } from './lib/utilities.js';
+import controllers from './controllers/index.js';
+
+/* istanbul ignore next */ if (['development', 'test'].includes(Settings.env)) dotenv.config();
 
 const app = express();
-const addons = require('./lib/addons');
-
-const { logger } = require('./lib/utilities');
 
-/* istanbul ignore next */ if (!global.__basedir) {
+/* istanbul ignore next */ if (!global?.__basedir) {
+  // eslint-disable-next-line no-global-assign
+  __dirname = dirname(fileURLToPath(import.meta.url));
```

Comment:
We probably should remove __dirname, no need to keep old and deprecated names. Also you could use `fileURLToPath(new URL('.', import.meta.url))`  which is the same as `dirname(fileURLToPath(import.meta.url))` except the new url method returns a trailing slash

Reply:
generally the `.` based urls have had issues on windows from my research

Reply:
i'll try it, but i'd rather remove it elsewhere, like in further cleanups. i'm just wanting to keep it consistent on this migration

Reply:
I see but this is the esm update, __dirname and __basedir is commonjs, i think this pr is the appropriate place to change it could be just another name it's not that big of a change, also i tried changing it, didn't show any weird behaviour

Reply:
i'll give it a look to see if i can do this better than how i am, but at a minimum i'll drop the global change and the underscore-prefixed name